api/agent/drivers/docker/docker.go

```diff
@@ -102,13 +146,34 @@ func NewDocker(conf drivers.Config) *DockerDriver {
 		}
 	}
 
+	go NewImageCleaner(driver, context.Background())
+
+	logrus.Infof("Max cache size: %v\n", conf.MaxImageCacheSize)
+	driver.imageCache = NewCache(int64(conf.MaxImageCacheSize))
+
+	imagesBeforeLoad, err := driver.docker.ListImages(context.Background())
+
 	if conf.DockerLoadFile != "" {
 		err = loadDockerImages(driver, conf.DockerLoadFile)
```

Comment:
it might be easier to modify loadDockerImages to get a list.. (And it would be nice to log the names of those images anyway.) then remove those from imagesBeforeLoad.